include/osquery/external_hash.h

```diff
@@ -0,0 +1,47 @@
+/*
```

Comment:
I'm modifying the code to use SQL calls to the hash table wherever possible.

However, there's still a number of places which are getting hashes directly from buffers instead of from files on the filesystem (the new implementation for this will use boost::uuids::detail::sha1 instead of OpenSSL).

In the buffer case, I think it would still be best to have a specific function to handle this. Such a function would make the most sense in a separate header, IMO, which at the moment is called external_hash.h

Would you prefer it to be implemented separately? I previously played around with the idea of defining this "sha1-from-buffer" function inline in hash.h, but that seems even less clean than having a separate header for it.